### DISCUSSION TITLE
Model-viewer editor outputs data-surface instead of data-position and data-normal, hotspots showing up in middle of model

### DISCUSSION THREAD

**Discussion by 5BYFIVE** [google/model-viewer#4184] [Q&A]
It's been a minute since I had to use the editor to output a GLTF and hotspots and I see a couple things are different. In my site, I updated the link to the newest version of the model-viewer JS. My old pages using data-position and data-normal are still working ok at rendering and displaying the hotspots. I'm adding a new page with a new GLTF with the data-surface parameter and the hotspots show in the middle of the model. Any ideas what I'm doing wrong? Thanks!

**This is working...**

`</model-viewer bounds="tight" src="/3dfile.gltf" enable-pan="" camera-controls="" interaction-prompt="none" oncontextmenu="return false;" environment-image="neutral" poster="poster.jpg" shadow-intensity="0" data-js-focus-visible="" ar-status="not-presenting">
	
		<button class="Hotspot" slot="hotspot-1" date-surface="" data-position="0.03696247474776085m 0.597153643488307m 0.005503934808075428m" data-normal="0m 0m 1m" data-visibility-attribute="visible" x-on:click="togglePopUp('example-1')" x-show="showLabels" x-transition="" data-visible="">
			<div class="HotspotAnnotation">Hotspot 1</div>
		</button>
</model-viewer>
`

**This is not working and displaying the hotspots in the center...**

`</model-viewer bounds="tight" src="/NEW_3dfile.gltf" enable-pan="" camera-controls="" interaction-prompt="none" oncontextmenu="return false;" environment-image="neutral" poster="poster.png" shadow-intensity="0" data-js-focus-visible="" ar-status="not-presenting">
	
		<button class="Hotspot" slot="hotspot-1" date-surface="0 0 600 625 624 0.481 0.410 0.109" data-visibility-attribute="visible" x-on:click="togglePopUp('example-1')" x-show="showLabels" x-transition="">
			<div class="HotspotAnnotation">Haptics</div>
		</button>
	
</model-viewer>`

**Comment by elalish** [comment]
Hmm, not entirely sure. Are you using the glTF you uploaded to the editor, or the one you downloaded from it? I could imagine some indices might have been mixed up on export, though it would still be a bug. Can you share a repro link? Any errors in the console?

**Reply by 5BYFIVE** [reply]
I'm using the glTF I downloaded from the editor. The packaged files from the editor work ok, so it must be something with my setup. I'll have to remove some more things to get to basics to diagnose. I did have errors in the console but discovered that was because I had empty data-position and data-normal parameters in there along with the data-surface parameter. No errors currently. Thanks for looking!

You can view the new trouble page at http://3dcanada.5byfivecreative.com/en/product/eyecee-one-iol/
A page that works with the old stuff is at http://3dcanada.5byfivecreative.com/en/product/envista-iol/

user: bl3d
pass: same as user

**Reply by elalish** [reply]
Can you try using the glTF you uploaded to the editor instead of the one you downloaded from it? I'm curious if that might fix it, and if so it'll point me straight to the problem.

**Reply by 5BYFIVE** [reply]
Unfortunately it did not fix it, will hopefully do more testing this afternoon

**Reply by 5BYFIVE** [reply]
Oh holy hell, I had a long reply typed out with all my tests and results and realized I had a typo in my code, I had "date-surface" instead of "data-surface"...everyhting seems to work now. I'm sorry for wasting your time!
